staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -462,6 +484,12 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 		})
 	}
 
+	// If user provided an IP address, ensure that IP address is found
+	// in the cloud provider before removing the taint on the node
+	if nodeIP, ok := ensureNodeProvidedIPExists(node, nodeAddresses); ok && nodeIP == nil {
```

Comment:
Why was this added here?

Reply:
Same logic of current line 445, because both instances and instancesV2 should handle nodeAddresses by ensureNodeProvidedIPExists.

Reply:
Do you mean it should be checked before `if instanceType != ""` ?

Reply:
I see, I misread the diff sorry :)